### PR TITLE
CHG: Set state to "idle" adter 300ms (fixes #23)

### LINKED ISSUE #23: expire status after x seconds
Using eventsensor to track motion sensor events from my netatmo camera.  
Currently the state stays in the last detected motion.

Similar to [MQTT sensor](https://www.home-assistant.io/integrations/sensor.mqtt/), is it possible to expire the status after a while?

This so the state goes back to `off` after e.g. `10 seconds`.

### PR DESCRIPTION
**Pull request by koying**
By definition, an event does not have a long-term state.
I have this issue that, e.g., I press "off" on a remote, then an automation turns on the lights, then press "off" on the remote again, an automation based on the state of an eventsensor won't trigger, because the state didn't actually changed.

This PR sets the long-term state to "idle" 300ms after an event.

### CODE REVIEW

**Comment by azogue** [comment]
Hi @koying, 

I see what your _specific_ problem is, but completely changing this custom integration behaviour is not the way :)

The main objective of this CC is **to track** HA events, making them _persistent_ over time. And doing that by setting the state of the EventSensor each time the event fires.
Your change breaks that and sets a fixed `"idle"` state in less than a second, and it's not even a config option, but a hardcoded one. And does not _fix_ #23, as the user wanted to _expire the sensor_ to an `"off"` state in ~10 sec, not to idle in 0.3s...

For your use case (which, btw, is already discussed in the forums), **what you really need is to use a _device trigger_ in your OFF-button automation**, which is a UI helper to set the automation trigger to the exact event that the off button of the remote fires in HA --> you don't need this custom integration :)

IF, anyway, you want to follow this course, which I suppose **it could be useful for other users** (most of them also ignoring the useful device triggers in HA Core), then **this MR is insufficient**, because:
- It changes the old behaviour (it should be a _new_ option, disabled by default)
- It needs to be configurable, for both the state to use when expired, and the time to do so.
- Configuration needs to happen in the config flow (a new last step, maybe?)
- It needs to be documented in README (and in the HA config flow)

With those points checked, I won't have any problems merging this, but can't do it as it is now.

Sorry for the late answer, btw, I just saw it. If not cc with @azogue I don't receive notifications :(

File: custom_components/eventsensor/sensor.py [hunk 1]
```diff
@@ -170,6 +178,7 @@ def async_update_sensor(event: Event):
                 }
                 _LOGGER.debug("%s: New state: %s", self.entity_id, self._state)
                 self.async_write_ha_state()
+                asyncio.ensure_future(self.async_set_idle())
```
Comment (azogue) [inline custom_components/eventsensor/sensor.py:181]: @koying,

In addition to the general comment about this PR, the implementation is also wrong... sorry :(

What happens if > 1 events are fired in a short period of time? :)

**Review by azogue — CHANGES_REQUESTED** [review]

File: custom_components/eventsensor/sensor.py (hunk 1 above)
Comment (koying) [reply custom_components/eventsensor/sensor.py:181]: I should cancel the future, right?
I'm really not a python guru ;)

**Comment by koying** [comment]
Thanks for the top regarding device trigger, but my use-case for using eventsensor in the first place is to "transfer" the events from one HA to another through MQTT, so that doesn't apply.

Re whether this should be pursued or not, I leave it to you, I'm perfectly ok with my fork for the foreseeable future.
I honestly didn't get to that the whole point of the integration was to *persist* events. If so, that PR makes no sense, of course :)

**Comment by azogue** [comment]
> Thanks for the tip regarding device trigger, but my use-case for using eventsensor in the first place is to "transfer" the events from one HA to another through MQTT, so that doesn't apply.

Strange use-case indeed :)
But I must insist, if you really want to publish a MQTT message when some event fires, you could entirely bypass the sensor thing... 
1. Remote linked to HA nº1 fires an event when button is pressed
2. Automation in HA 1, triggered with device trigger or by plain event data, **publishes the MQTT msg** (it's a service like any other)
3. HA nº2, which is connected to the same MQTT broker I assume, listens to the topic and does its thing

Is it not a valid workflow for your scenario?

> I honestly didn't get to that the whole point of the integration was to _persist_ events. If so, this PR makes no sense, of course :)

Thanks for understanding 🍻

**Comment by koying** [comment]
> Is it not a valid workflow for your scenario?

Probably, but I already automatized the publishing of generic sensors (amongst other) between HA1 and HA2, so eventsensor basically already implements all the boilerplate I would have to code in automations for every single zigbee remotes I could have now or in the future.  

An having a history/log of the "events" is a big plus, ofc.

File: custom_components/eventsensor/sensor.py (hunk 1 above)
Comment (azogue) [reply custom_components/eventsensor/sensor.py:181]: yes, is not a good thing to fire those without control.

There are already ways to do that kind of things in HA Core, and it is better to use them. For example, in `homeassistant.helpers.event` there are some helpers to trigger some action after some time. You could explore their usages in the HA Core code, for inspiration or just to copycat how to do it

**Comment by azogue** [comment]
> > Is it not a valid workflow for your scenario?
> 
> Probably, but I already automatized the publishing of generic sensors (amongst other) between HA1 and HA2, so eventsensor basically already implements all the boilerplate I would have to code in automations for every single zigbee remotes I could have now or in the future.
> 
> An having a history/log of the "events" is a big plus, ofc.

Well, do what you want, of course, but I'm not confident about that _reset_ of the event sensor after 300ms, I think it will generate issues the moment you receive quick bursts of events... 

An automation with a trigger after some inactivity time over the eventsensor would work much better, for multiple reasons:
- The logic of waiting for inactivity, canceling the reset if another event comes, etc. is already implemented there, and with a much more robust way of working. And, recently, there are ways to configure automations to adapt to _trigger bursts_, by configuring the [run mode](https://www.home-assistant.io/docs/automation/modes/) 
- With the new _blueprints_ you could do some automation schema valid for all your remotes, and then use it for all just changing the entity name (I didn't try it yet, but I suppose is a better workflow than copy-paste a lot of automations)

Anyway, if you have invested a lot of time in your setup with those 2 HA instances coupled with a MQTT broker, I understand that you prefer to use your custom-tunned EventSensor, so feel free to do it :)
But I cannot add these changes to the repo as they are, I hope you understand me

**Comment by koying** [comment]
>  An automation with a trigger after some inactivity time over the eventsensor [...] The logic of waiting for inactivity, canceling the reset if another event comes, etc

Mmm, not sure exactly what you have in mind.
If you would be so kind to elaborate, I'm all for using the "stock" eventsensor rather than a fork :)

**Pull request closed without merging**